File: Chapter3.CoffeeShop/Models/Condiments/Mocha.cs
namespace Chapter3.CoffeeShop.Models;

public class Mocha : CondimentDecorator
{
    public Mocha(Beverage beverage)
    {
        _beverage = beverage;
    }


    public override double Cost()
    {
        var price = 0.20;

        var cost = _beverage.GetSize() switch
        {
            Size.Tall => price - (price / 3),
            Size.Grande => price,
            Size.Venti => price + (price / 3),
            _ => throw new ArgumentOutOfRangeException()
        };

        return _beverage.Cost() + cost;
    }

// public override double Cost() => _beverage.Cost() + 0.20;
    public override string GetDescription() => _beverage.GetDescription() + ", Mocha";
}

File: Chapter3.CoffeeShop.Java/src/Condiments/Mocha.java
package Condiments;

import Beverages.Beverage;

public class Mocha extends CondimentDecorator {

    public Mocha(Beverage beverage) {
        this.beverage = beverage;
    }

    @Override
    public double cost() {
        Double pricing = switch (beverage.getSize()) {
            case TALL -> 0.15;
            case GRANDE -> 0.20;
            case VENTI -> 0.25;
        };
        return beverage.cost() + pricing;
    }

    @Override
    public String getDescription() {
        return beverage.getDescription() + ", Mocha";
    }
}
